File: backend/src/main/java/com/nusmotion/backend/dto/Announcement.java
package com.nusmotion.backend.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.OffsetDateTime;

/**
 * Service announcement from /Announcements.
 */
public record Announcement(
        @JsonProperty("ID") String id,
        @JsonProperty("Text") String text,
        @JsonProperty("Status") String status,
        @JsonProperty("Priority") String priority,
        @JsonProperty("Affected_Service_Ids") String affectedServiceIds,
        @JsonProperty("Created_On") OffsetDateTime createdOn,
        @JsonProperty("Created_By") String createdBy
) {}

File: frontend/lib/data/models/announcement.dart
class Announcement {
  final String id;
  final String text;
  final String status;
  final String priority;
  final String affectedServiceIds;

  /// Timestamp when the alert was created (from backend "createdOn").
  final DateTime? createdOn;

  /// Who created the alert (from backend "createdBy").
  final String? createdBy;

  const Announcement({
    required this.id,
    required this.text,
    required this.status,
    required this.priority,
    required this.affectedServiceIds,
    this.createdOn,
    this.createdBy,
  });

  factory Announcement.fromJson(Map<String, dynamic> json) {
    DateTime? parsedCreatedOn;
    final createdOnRaw = json['Created_On'];
    if (createdOnRaw != null) {
      parsedCreatedOn = DateTime.tryParse(createdOnRaw.toString());
    }

    return Announcement(
      id: json['ID'] as String? ?? '',
      text: json['Text'] as String? ?? '',
      status: json['Status'] as String? ?? '',
      priority: json['Priority'] as String? ?? '',
      affectedServiceIds: json['Affected_Service_Ids'] as String? ?? '',
      createdOn: parsedCreatedOn,
      createdBy: json['Created_By'] as String?,
    );
  }
}
